src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -15,20 +22,26 @@ const WAVEFROM_SCALE_FACTOR = 5
 let recordingLength = 0;
 let recordBuffer = [[], []];
 let isRecording = false;
-let isMonitoring = false;
-let visualizationEnabled = true;
+let recordingState = recorderState.UNINITIALIZED;
 
 let recordButton = document.querySelector('#record');
 let recordText = document.querySelector('#record-text');
+let stopButton = document.querySelector('#stop');
 let player = document.querySelector('#player');
 let downloadButton = document.querySelector('#download-button');
 let downloadLink = document.querySelector('#download-link');
 
 // Wait for user interaction to initialize audio, as per specification.
-document.addEventListener('click', (element) => {
-  init();
-  document.querySelector('#click-to-start').remove();
-}, {once: true});
+if (recordingState === recorderState.UNINITIALIZED) {
```

Comment:
Is this `if` necessary? This code will get executed because `recordingState` is always `UNINITIALIZED` at the time of execution.

Reply:
Yeah, it is not necessary, but if we remove this "if statement", then we don't need enum for SPN anymore. It will make the SPN recorder and worklet recorder code is not consistent. The reader might got confuse why one example use enum, but the other one don't use enum without deeply reading the our demo code.